trin-core/src/types/accumulator.rs

```diff
@@ -116,45 +82,36 @@ impl MasterAccumulator {
         Ok(epoch_acc[rel_index as usize].block_hash)
     }
 
-    fn is_header_in_current_epoch(&self, header: &Header) -> bool {
-        let current_epoch_block_min = match self.historical_epochs_header_count() {
-            0 => match header.number {
-                0u64 => return true,
-                _ => 0,
-            },
-            _ => self.historical_epochs_header_count() - 1,
+    pub fn validate_header_with_proof(&self, hwp: &HeaderWithProof) -> anyhow::Result<()> {
+        let proof = match &hwp.proof {
+            BlockHeaderProof::AccumulatorProof(val) => {
+                if hwp.header.number > MERGE_BLOCK_NUMBER {
+                    return Err(anyhow!("Invalid proof type found for post-merge header."));
+                }
+                val
+            }
+            BlockHeaderProof::None(_) => {
+                if hwp.header.number <= MERGE_BLOCK_NUMBER {
+                    return Err(anyhow!("Missing accumulator proof for pre-merge header."));
+                } else {
+                    // Skip validation for post-merge headers until proof format is finalized
+                    return Ok(());
+                }
+            }
         };
```

Comment:
nit: we could clean this up by rearranging the conditional statements

example:

```rust
if hwp.header.number > MERGE_BLOCK_NUMBER {
    if let BlockHeaderProof::AccumulatorProof(..) = hwp.proof {
        return Err(..);
    }
    return Ok(());
}

if let BlockHeaderProof::AccumulatorProof(proof) = hwp.proof {
    // validation...
} else {
    Err(..)
}
```

Reply:
I don't think it's necessary, we're going to have some nested logic regardless (in your example, nesting all of the validation logic.